### DISCUSSION TITLE
Lead/HTML forms 📝

### DISCUSSION THREAD

**Discussion by FelixMalfait** [twentyhq/twenty#207] [Ideas]
On Twenty.com, we have a form to collect leads but it is not possible yet to automatically send the content of that form to the CRM.

It would be nice to have a FormBuilder that allows us to create forms to fill internal objects from external sources. 
We probably want to set enough a abstraction initially to be able to get to that "form builder" vision later, but ship a v1 that is limited to filling contact/company object.

Since the form endpoint will be available externally without auth, it should implement a captcha solution.

**Comment by Marioviki** [comment]
Is this builder can be used for such shop Viking jewelry, rings, bracelet?["https://vikingthunder.com/"](url)

**Comment by B2VMarketing** [comment]
Lead forms would be a great addition, even a Google Forms connector would be sufficient.

**Comment by FelixMalfait** [comment]
Note we now have Workflows with Webhook triggers in Workflows so it's very easy to plug an external form system.

**Reply by hubyhuby** [reply]
> Note we now have Workflows with Webhook triggers in Workflows so it's very easy to plug an external form system.

Just fiddling around to try to understand how easy it is to get my contact form into twenty CRM... It doesn t look as easy as with make.com for example... Where you just select your google account, then your excel spreadsheet. I understand we don t have a make or n8n directly in twenty (for now ;) but I don t see the easy part in "easy to plug an external form system" :D  

I imagine now that we have the Apps in Twenty 2.X , that the idea is to develop an App that would do this work for connecting to google form, or create a simple contact form like on Calendly ( name email phone and drop downs to qualify the leads). 

<img width="1817" height="1053" alt="image" src="https://github.com/user-attachments/assets/f6076089-e6f8-4116-afb6-692cee5e9bce" />

**Reply by FelixMalfait** [reply]
What you did above is a good start, but it should be a POST request not a GET so that you can map the payload.

With apps what could be done is directly building the form and serving it with Twenty (no Google Form at all, you create a publicly exposed Logic Function that serves the HTML), but that's more advanced 😅

**Comment by jevtih-svg** [comment]
This is so easily resolved with simple script on the website side, especially with the AI...

**Reply by hubyhuby** [reply]
We finally used Formbricks and connected it with custom code. No webhooks like you said.

**Comment by FelixMalfait** [comment]
Now with AI + apps it's also easy to publish a mini "website form". It's not well documented but we'll soon make it more obvious in the app how to do that. Within a couple month it should be doable in the AI chat "build a small landing page with a form that asks for X/Y and store it in object Z" then it will handle everything
